### DISCUSSION TITLE
Loss when using Alignment Network in parallel models(ie. FastPitch)

### DISCUSSION THREAD

**Discussion by talipturkmen** [coqui-ai/TTS#807] [General Q&A]
As far as I understand from the implementation of FastPitch when using the alignment network since we are using predicted durations instead of ground truth durations. I noticed that in the loss of fastpitch we use also MSE loss over predicted melspec and ground truth melspec. If we use real alignments it is ok. However, in the case of usage of alignment network shouldn't we need a DTW loss. Since we are predicting durations our alignments will not match the real alignments and the reconstruction loss seems wrong to me. Is it a bug or am I missing something?

**Comment by erogol** [comment]
No, we don't need DTW (but DTW can be an improvement). The alignment network is similar to the attention module used in Tacotron. It learns how the alignment between ground truth spectrograms and the text. Therefore in training, there is no misalignment. However, if we train the model only using the duration predictor then I'd guess we'd need the DTW loss. 

Hope it makes sense ?

**Comment by erogol** [comment]
Moving this to discussions as it fits there better.
